### ISSUE TITLE
⚠️ FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.   offset = pd.tseries.frequencies.to_offset(freq)

### ISSUE THREAD

**Issue by galenseilis** [sktime/sktime#6694]
**Describe the bug**

This is more of a warning about the future state rather than a bug in the current state.

```bash
$ python example1.py 
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/utils/datetime.py:73: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  offset = pd.tseries.frequencies.to_offset(freq)
```

**To Reproduce**
Following [this example](https://www.sktime.net/en/latest/examples/01b_forecasting_proba.html#Probabilistic-forecasting-interfaces-in-sktime)

```python
import numpy as np

from sktime.datasets import load_airline
from sktime.forecasting.theta import ThetaForecaster

# until fit, identical with the simple workflow
y = load_airline()

fh = np.arange(1, 13)

forecaster = ThetaForecaster(sp=12)
forecaster.fit(y, fh=fh)
```

**Expected behavior**

No warning.

**Additional context**

None.

**Versions**
<details>

```python
>>> from sktime import show_versions; show_versions()

System:
    python: 3.12.3 (main, Apr 15 2024, 18:25:56) [Clang 17.0.6 ]
executable: /home/galen/projects/try-sktime/.venv/bin/python3
   machine: Linux-5.15.0-112-generic-x86_64-with-glibc2.35

Python dependencies:
          pip: None
       sktime: 0.30.1
      sklearn: 1.5.0
       skbase: 0.8.1
        numpy: 1.26.4
        scipy: 1.14.0
       pandas: 2.2.2
   matplotlib: 3.9.0
       joblib: 1.4.2
        numba: None
  statsmodels: 0.14.2
     pmdarima: None
statsforecast: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None
tensorflow_probability: None
```

```bash
$ rye make-req
$ cat requirements.lock 
# generated by rye
# use `rye lock` or `rye sync` to update this lockfile
#
# last locked with the following flags:
#   pre: false
#   features: []
#   all-features: false
#   with-sources: false
#   generate-hashes: false

-e file:.
contourpy==1.2.1
    # via matplotlib
cycler==0.12.1
    # via matplotlib
fonttools==4.53.0
    # via matplotlib
joblib==1.4.2
    # via scikit-learn
    # via sktime
kiwisolver==1.4.5
    # via matplotlib
matplotlib==3.9.0
    # via seaborn
    # via try-sktime
numpy==1.26.4
    # via contourpy
    # via matplotlib
    # via pandas
    # via patsy
    # via scikit-learn
    # via scipy
    # via seaborn
    # via sktime
    # via statsmodels
packaging==24.1
    # via matplotlib
    # via sktime
    # via statsmodels
pandas==2.2.2
    # via seaborn
    # via sktime
    # via statsmodels
patsy==0.5.6
    # via statsmodels
pillow==10.3.0
    # via matplotlib
pyparsing==3.1.2
    # via matplotlib
python-dateutil==2.9.0.post0
    # via matplotlib
    # via pandas
pytz==2024.1
    # via pandas
scikit-base==0.8.1
    # via sktime
scikit-learn==1.5.0
    # via sktime
scipy==1.14.0
    # via scikit-learn
    # via sktime
    # via statsmodels
seaborn==0.13.2
    # via try-sktime
six==1.16.0
    # via patsy
    # via python-dateutil
sktime==0.30.1
    # via try-sktime
statsmodels==0.14.2
    # via try-sktime
threadpoolctl==3.5.0
    # via scikit-learn
tzdata==2024.1
    # via pandas
```

</details>

**Comment by galenseilis** [comment]
I also tried running a probabilistic forecast, which gave the same warning:

```python
>>> yhat = forecaster.predict_proba()
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:400: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  if freq_from_self != freq_from_obj:
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:172: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  return to_offset(obj)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:400: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  if freq_from_self != freq_from_obj:
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:172: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  return to_offset(obj)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/utils/datetime.py:73: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  offset = pd.tseries.frequencies.to_offset(freq)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:172: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  return to_offset(obj)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:400: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  if freq_from_self != freq_from_obj:
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:172: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  return to_offset(obj)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/utils/datetime.py:73: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  offset = pd.tseries.frequencies.to_offset(freq)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:172: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  return to_offset(obj)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:400: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  if freq_from_self != freq_from_obj:
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:172: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  return to_offset(obj)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/utils/datetime.py:73: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  offset = pd.tseries.frequencies.to_offset(freq)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:400: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  if freq_from_self != freq_from_obj:
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:172: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  return to_offset(obj)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/utils/datetime.py:73: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  offset = pd.tseries.frequencies.to_offset(freq)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:172: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  return to_offset(obj)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:400: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  if freq_from_self != freq_from_obj:
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:172: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  return to_offset(obj)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/utils/datetime.py:73: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  offset = pd.tseries.frequencies.to_offset(freq)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:172: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  return to_offset(obj)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:400: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  if freq_from_self != freq_from_obj:
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/forecasting/base/_fh.py:172: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  return to_offset(obj)
/home/galen/projects/try-sktime/.venv/lib/python3.12/site-packages/sktime/utils/datetime.py:73: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  offset = pd.tseries.frequencies.to_offset(freq)
```

**Comment by fkiraly** [comment]
Duplicate of https://github.com/sktime/sktime/issues/6245?

Also see https://github.com/sktime/sktime/issues/6572

May be blocked by pandas bug https://github.com/pandas-dev/pandas/issues/58974

**Comment by fkiraly** [comment]
these warnings from `pandas 2.2` are really aggressive - we have upper bounds on pandas, so perhaps it is best to silence them until https://github.com/sktime/sktime/issues/6245 is done

**Comment by spillz** [comment]
Could I get some clarity on the intended direction of pandas development here?

For `Q-DEC` I get deprecation warnings telling me to use `QE-DEC` instead (same for `Q` and `QE`, `M-DEC` and `ME-DEC`, etc). But then I try to use that same end frequency in `Period` indexed data and I get errors to the effect that ends are not supported. I view `Period` as essential pandas functionality in time series economics applications supporting symmetric frequency definitions in `Timestamp` and `Period` simplifies a lot of logic.

**Comment by fkiraly** [comment]
@spillz, we are not developing `pandas`, so I am unsure where this will be going. So far, the fixes in `sktime` have been reactive to changes in `pandas`, and we will probably be moving the forecasting horizons away from `pandas` to have a more stable object.
